src/common/snippets/src/lowered/pass/pass.cpp

```diff
@@ -16,23 +16,33 @@ PassPipeline::PassPipeline(const std::shared_ptr<PassConfig>& pass_config) : m_p
     OPENVINO_ASSERT(m_pass_config != nullptr, "PassConfig is not initialized!");
 }
 
-void PassPipeline::register_pass(const snippets::pass::PassPosition& position, const std::shared_ptr<Pass>& pass) {
+void PassPipeline::register_pass(const snippets::pass::PassPosition& position, const std::shared_ptr<PassBase>& pass) {
     OPENVINO_ASSERT(pass != nullptr, "PassPipeline cannot register empty pass!");
     m_passes.insert(position.get_insert_position(m_passes), pass);
 }
 
-void PassPipeline::register_pass(const std::shared_ptr<Pass>& pass) {
+void PassPipeline::register_pass(const std::shared_ptr<PassBase>& pass) {
     OPENVINO_ASSERT(pass != nullptr, "PassPipeline cannot register empty pass!");
     m_passes.push_back(pass);
 }
 
 void PassPipeline::run(LinearIR& linear_ir) const {
+    run(linear_ir, linear_ir.cbegin(), linear_ir.cend());
+}
+
+void PassPipeline::run(LinearIR& linear_ir, LinearIR::constExprIt begin, LinearIR::constExprIt end) const {
     for (const auto& pass : m_passes) {
         OPENVINO_ASSERT(pass != nullptr, "PassPipeline has empty pass!");
         if (m_pass_config->is_disabled(pass->get_type_info())) {
             continue;
         }
-        pass->run(linear_ir);
+        if (auto lir_pass = std::dynamic_pointer_cast<Pass>(pass)) {
+            lir_pass->run(linear_ir);
+        } else if (auto ranged_pass = std::dynamic_pointer_cast<RangedPass>(pass)) {
+            ranged_pass->run(linear_ir, begin, end);
```

Comment:
If we felt a necessity of the running several ranged passes on different ranges in a single `PassPipeline`, we could add a new API to a base `RangedPass` class (e.g. methods `define_ranges` and `ranges_are_defined`) and use it in `PassPipeline::run` overloads.
However, I can't imagine a situation in which this would be needed